python/segyio/segyio.cpp

```diff
@@ -2772,6 +2780,127 @@ extern "C" int segy_parse_layout_xml(
     return SEGY_OK;
 }
 
+int entry_defintion_to_py_TraceHeaderLayoutEntry(
+    PyObject* entries,
+    int offset,
+    const segy_entry_definition& def,
+    PyObject* entry_class
+) {
+    size_t entry_type_map_size = sizeof( entry_type_map ) / sizeof( entry_type_map[0] );
+
+    std::string entry_type;
+    for( size_t i = 0; i < entry_type_map_size; ++i ) {
+        if( def.entry_type == entry_type_map[i].segyio_entry_type ) {
+            entry_type = entry_type_map[i].spec_entry_type;
+            break;
+        }
+    }
```

Comment:
If this is actually too expensive, search for every entry type, I guess the simplest is to set up some lookup table/method to avoid the search.